CI failing with SciPy 1.8.1 release

## Describe the bug
The test suite fails to run with the following error:

``` python
Traceback (most recent call last):
  File "/usr/share/miniconda/bin/pytest", line 11, in <module>
    sys.exit(console_main())
  File "/usr/share/miniconda/lib/python3.10/site-packages/_pytest/config/__init__.py", line 1[8](https://github.com/jacksund/simmate/runs/6538445803?check_suite_focus=true#step:7:9)7, in console_main
    code = main()
  File "/usr/share/miniconda/lib/python3.10/site-packages/_pytest/config/__init__.py", line 145, in main
    config = _prepareconfig(args, plugins)
  File "/usr/share/miniconda/lib/python3.10/site-packages/_pytest/config/__init__.py", line 324, in _prepareconfig
    config = pluginmanager.hook.pytest_cmdline_parse(
  File "/usr/share/miniconda/lib/python3.10/site-packages/pluggy/_hooks.py", line 265, in __call__
    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
  File "/usr/share/miniconda/lib/python3.10/site-packages/pluggy/_manager.py", line 80, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/share/miniconda/lib/python3.10/site-packages/pluggy/_callers.py", line 55, in _multicall
    gen.send(outcome)
  File "/usr/share/miniconda/lib/python3.10/site-packages/_pytest/helpconfig.py", line 102, in pytest_cmdline_parse
    config: Config = outcome.get_result()
  File "/usr/share/miniconda/lib/python3.10/site-packages/pluggy/_result.py", line 60, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/usr/share/miniconda/lib/python3.10/site-packages/pluggy/_callers.py", line 3[9](https://github.com/jacksund/simmate/runs/6538445803?check_suite_focus=true#step:7:10), in _multicall
    res = hook_impl.function(*args)
  File "/usr/share/miniconda/lib/python3.[10](https://github.com/jacksund/simmate/runs/6538445803?check_suite_focus=true#step:7:11)/site-packages/_pytest/config/__init__.py", line 1016, in pytest_cmdline_parse
    self.parse(args)
  File "/usr/share/miniconda/lib/python3.10/site-packages/_pytest/config/__init__.py", line 1304, in parse
    self._preparse(args, addopts=addopts)
  File "/usr/share/miniconda/lib/python3.10/site-packages/_pytest/config/__init__.py", line 1206, in _preparse
    self.hook.pytest_load_initial_conftests(
  File "/usr/share/miniconda/lib/python3.10/site-packages/pluggy/_hooks.py", line 265, in __call__
    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
  File "/usr/share/miniconda/lib/python3.10/site-packages/pluggy/_manager.py", line 80, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/share/miniconda/lib/python3.10/site-packages/pluggy/_callers.py", line 60, in _multicall
    return outcome.get_result()
  File "/usr/share/miniconda/lib/python3.10/site-packages/pluggy/_result.py", line 60, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/usr/share/miniconda/lib/python3.10/site-packages/pluggy/_callers.py", line 39, in _multicall
    res = hook_impl.function(*args)
  File "/usr/share/miniconda/lib/python3.10/site-packages/pytest_django/plugin.py", line 353, in pytest_load_initial_conftests
    _setup_django()
  File "/usr/share/miniconda/lib/python3.10/site-packages/pytest_django/plugin.py", line 236, in _setup_django
    django.setup()
  File "/usr/share/miniconda/lib/python3.10/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/share/miniconda/lib/python3.10/site-packages/django/apps/registry.py", line [11](https://github.com/jacksund/simmate/runs/6538445803?check_suite_focus=true#step:7:12)6, in populate
    app_config.import_models()
  File "/usr/share/miniconda/lib/python3.10/site-packages/django/apps/config.py", line 304, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/share/miniconda/lib/python3.10/importlib/__init__.py", line [12](https://github.com/jacksund/simmate/runs/6538445803?check_suite_focus=true#step:7:13)6, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/runner/work/simmate/simmate/src/simmate/website/core_components/models.py", line 7, in <module>
    from simmate.database.base_data_types import Spacegroup
  File "/home/runner/work/simmate/simmate/src/simmate/database/base_data_types/__init__.py", line 15, in <module>
    from .static_energy import StaticEnergy
  File "/home/runner/work/simmate/simmate/src/simmate/database/base_data_types/static_energy.py", line [13](https://github.com/jacksund/simmate/runs/6538445803?check_suite_focus=true#step:7:14), in <module>
    from pymatgen.io.vasp.outputs import Vasprun
  File "/usr/share/miniconda/lib/python3.10/site-packages/pymatgen/io/vasp/__init__.py", line 12, in <module>
    from .outputs import (
  File "/usr/share/miniconda/lib/python3.10/site-packages/pymatgen/io/vasp/outputs.py", line 45, in <module>
    from pymatgen.electronic_structure.dos import CompleteDos, Dos
  File "/usr/share/miniconda/lib/python3.10/site-packages/pymatgen/electronic_structure/dos.py", line [16](https://github.com/jacksund/simmate/runs/6538445803?check_suite_focus=true#step:7:17), in <module>
    from scipy.signal import hilbert
  File "/usr/share/miniconda/lib/python3.10/site-packages/scipy/signal/__init__.py", line [30](https://github.com/jacksund/simmate/runs/6538445803?check_suite_focus=true#step:7:31)9, in <module>
    from . import _sigtools, windows
  File "/usr/share/miniconda/lib/python3.10/site-packages/scipy/signal/windows/__init__.py", line 41, in <module>
    from ._windows import *
  File "/usr/share/miniconda/lib/python3.10/site-packages/scipy/signal/windows/_windows.py", line 7, in <module>
    from scipy import linalg, special, fft as sp_fft
  File "/usr/share/miniconda/lib/python3.10/site-packages/scipy/fft/__init__.py", line 91, in <module>
    from ._helper import next_fast_len
  File "/usr/share/miniconda/lib/python3.10/site-packages/scipy/fft/_helper.py", line 3, in <module>
    from ._pocketfft import helper as _helper
  File "/usr/share/miniconda/lib/python3.10/site-packages/scipy/fft/_pocketfft/__init__.py", line 3, in <module>
    from .basic import *
  File "/usr/share/miniconda/lib/python3.10/site-packages/scipy/fft/_pocketfft/basic.py", line 6, in <module>
    from . import pypocketfft as pfft
ImportError: /usr/share/miniconda/lib/python3.10/site-packages/pandas/_libs/window/../../../../../libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /usr/share/miniconda/lib/python3.10/site-packages/scipy/fft/_pocketfft/pypocketfft.cpython-[31](https://github.com/jacksund/simmate/runs/6538445803?check_suite_focus=true#step:7:32)0-x86_[64](https://github.com/jacksund/simmate/runs/6538445803?check_suite_focus=true#step:7:65)-linux-gnu.so)
Error: Process completed with exit code 1.
```


## To Reproduce
Bug is only present in the CI (github actions workflow). When running locally, there is no issue

## Additional context
- this error lines up with the new scipy release in the last 24hrs (see [conda forge](https://anaconda.org/conda-forge/scipy/files))
- I cannot reproduce this error locally with a brand new environment
- our github action installs our enviornment to `base` which uses the anaconda channel (instead of conda-forge). I probably need to switch our CI to create a brand new env to avoid errors like this

--- a/.github/workflows/lint-and-test.yml
+++ b/.github/workflows/lint-and-test.yml
@@ -4,6 +4,7 @@
 # 
 # We also use a third-party action for setting up our conda environment. See:
 #   https://github.com/marketplace/actions/setup-miniconda
+#
 # OPTIMIZE: consider switching to mamba for faster CI installs:
 #   https://github.com/marketplace/actions/setup-miniconda#example-6-mamba
 
@@ -22,8 +23,11 @@ jobs:
   build-linux:
     runs-on: ubuntu-latest
     steps:
+
     - uses: actions/checkout@v2
+
     - uses: psf/black@stable
+
     - name: Create conda env and install deps
       uses: conda-incubator/setup-miniconda@v2
       with:
@@ -32,14 +36,24 @@ jobs:
         auto-activate-base: false
         channels: conda-forge
         activate-environment: simmate_dev
+    # The lines "shell: bash -l {0}" ensure we use the created env below
+
     - name: Install Simmate
       shell: bash -l {0}
       run: |
-        conda info
         pip install -e .
+
+    - name: Check conda info
+      shell: bash -l {0}
+      run: |
+        conda info
+        conda list
+
     - name: Run default tests
+      shell: bash -l {0}
       run: |
         pytest
+
     # Marked tests can be dropped for now. I'll re-add this test when needed.
     # - name: Run no_django_setup tests
     #   run: |

--- a/.github/workflows/lint-and-test.yml
+++ b/.github/workflows/lint-and-test.yml
@@ -28,7 +28,7 @@ jobs:
       uses: conda-incubator/setup-miniconda@v2
       with:
         environment-file: tutorials/Guides_for_contributors/environment.yaml
-        python-version: 3.10
+        python-version: "3.10"
         auto-activate-base: false
         channels: conda-forge
         activate-environment: simmate_dev

--- a/.github/workflows/lint-and-test.yml
+++ b/.github/workflows/lint-and-test.yml
@@ -33,7 +33,9 @@ jobs:
         channels: conda-forge
         activate-environment: simmate_dev
     - name: Install Simmate
+      shell: bash -l {0}
       run: |
+        conda info
         pip install -e .
     - name: Run default tests
       run: |

--- a/.github/workflows/lint-and-test.yml
+++ b/.github/workflows/lint-and-test.yml
@@ -20,19 +20,32 @@ on:
     - cron:  '30 5,17 * * *'
 
 jobs:
-  build-linux:
-    runs-on: ubuntu-latest
+  lint-and-test:
+
+    # Establishes the combination of all OS and python versions
+    # for us to test. The "steps" below will be ran for each
+    # of these combinations. 
+    name: Test (${{ matrix.python-version }}, ${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        python-version: ["3.10"]
+
     steps:
 
-    - uses: actions/checkout@v2
+    - name: Initial setup
+      uses: actions/checkout@v2
 
-    - uses: psf/black@stable
+    - name: Check code format with Black
+      uses: psf/black@stable
 
     - name: Create conda env and install deps
       uses: conda-incubator/setup-miniconda@v2
       with:
         environment-file: tutorials/Guides_for_contributors/environment.yaml
-        python-version: "3.10"
+        python-version: ${{ matrix.python-version }}
         auto-activate-base: false
         channels: conda-forge
         activate-environment: simmate_dev

--- a/.github/workflows/lint-and-test.yml
+++ b/.github/workflows/lint-and-test.yml
@@ -1,6 +1,11 @@
 
 # This workflow will check for Black formating, install Python dependencies using Anaconda, and then run tests with a single version of Python.
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+# 
+# We also use a third-party action for setting up our conda environment. See:
+#   https://github.com/marketplace/actions/setup-miniconda
+# OPTIMIZE: consider switching to mamba for faster CI installs:
+#   https://github.com/marketplace/actions/setup-miniconda#example-6-mamba
 
 name: Run linting and tests with Anaconda
 
@@ -19,23 +24,21 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - uses: psf/black@stable
-    - name: Set up Python 3.10
-      uses: actions/setup-python@v2
+    - name: Create conda env and install deps
+      uses: conda-incubator/setup-miniconda@v2
       with:
-        python-version: "3.10"
-    - name: Add conda to system path
+        environment-file: tutorials/Guides_for_contributors/environment.yaml
+        python-version: 3.10
+        auto-activate-base: false
+        channels: conda-forge
+        activate-environment: simmate_dev
+    - name: Install Simmate
       run: |
-        # $CONDA is an environment variable pointing to the root of the miniconda directory
-        echo $CONDA/bin >> $GITHUB_PATH
-    - name: Install dependencies
-      run: |
-        conda update -n base -c defaults conda
-        conda env update -f tutorials/Guides_for_contributors/environment.yaml -n base
         pip install -e .
-    - name: Run unmarked tests
+    - name: Run default tests
       run: |
         pytest
-    # Marked tests can be dropped for now. I'll readd this test when needed.
+    # Marked tests can be dropped for now. I'll re-add this test when needed.
     # - name: Run no_django_setup tests
     #   run: |
     #     pytest -m no_django_setup -p no:django

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,7 @@ There is one key exception to the rules above -- and that is with `MAJOR`=0 rele
 
 # Upcoming Release
 > :bulb: for ongoing changes that have not been finalized/merged yet, view our [active pull-requests](https://github.com/jacksund/simmate/pulls) on github
+- update CI to test all OSs and pin pytest<7.1 as temporary fix for [#162](https://github.com/jacksund/simmate/issues/162)
 - fix spelling typos in `keyword_modifiers` ([@laurenmm](https://github.com/laurenmm), [#165](https://github.com/jacksund/simmate/pull/165))
 - users can now apply their own unique keyword modifiers to Incars -- such as how we allow "__per_atom" or "__smart_ismear" tags on Incar settings. This change involved refactoring how `keyword_modifiers` are implemented for the `vasp.inputs.Incar` class. Rather than static methods attached to the base class, modifiers are now dynamically applied using the `add_keyword_modifier` classmethod.
 - large update of `calculators.vasp.tasks` module where many new presets are reimplemented from [`pymatgen.io.vasp.sets`](https://pymatgen.org/pymatgen.io.vasp.sets.html). This includes robust unit testing to confirm that generated inputs match between simmate and pymatgen (see [#157](https://github.com/jacksund/simmate/issues/157) for a list of presets)

--- a/setup.py
+++ b/setup.py
@@ -125,7 +125,7 @@ setup(
         "pdoc>=11.0.0",  # for docs and markdown rendering in html templates
         #
         # For development and testing
-        "pytest>=6.2.5",
+        "pytest>=6.2.5,<7.1",  # BUG: see issue #162 for limiting <7.1
         "pytest-django>=4.5.2",
         "pytest-mock>=3.7.0",
         "pytest-xdist>=2.5.0",

--- a/tutorials/Guides_for_contributors/README.md
+++ b/tutorials/Guides_for_contributors/README.md
@@ -48,7 +48,6 @@ pip install -e .
 ```
 
 6. Make sure everything works properly by running our tests
-> :warning::warning::warning: there is currently a bug on windows for pytest v7.1 see issue [#162](https://github.com/jacksund/simmate/issues/162). Downgrade to v7.0 to fix this: `conda install -c conda-forge pytest==7.0`
 ``` shell
 # you can optionally run tests in parallel 
 # with a command such as "pytest -n 4"
@@ -83,7 +82,6 @@ black .
 > :bulb: you can optionally format files as you code with Spyder too. `Tools` -> `Preferences` -> `Completion and Linting` -> `Code Style and Formatting` > Under code formatting dropdown, select `black`. To format a file you have open in Spyder, use the the `Ctrl+Shift+I` shortcut.
 
 7. Simmate has test cases to make sure new changes don't break any of Simmate's existing features. These are written using [pytest](https://docs.pytest.org/en/6.2.x/). Run these to check your changes. Make sure you are in the `~/Documents/github/simmate` directory when you run this command:
-> :warning::warning::warning: there is currently a bug on windows for pytest v7.1 see issue [#162](https://github.com/jacksund/simmate/issues/162). Downgrade to v7.0 to fix this: `conda install -c conda-forge pytest==7.0`
 ``` shell
 # you can optionally run tests in parallel 
 # with a command such as "pytest -n 4"

--- a/tutorials/Guides_for_contributors/environment.yaml
+++ b/tutorials/Guides_for_contributors/environment.yaml
@@ -29,7 +29,7 @@ dependencies:
     - prefect >=0.15.11
     - pymatgen >=2022.1.9
     - pymatgen-analysis-diffusion >=2021.4.29
-    - pytest >=6.2.5
+    - pytest >=6.2.5,<7.1
     - pytest-django >=4.5.2
     - python >=3.10
     - pytest-mock >=3.7.0

